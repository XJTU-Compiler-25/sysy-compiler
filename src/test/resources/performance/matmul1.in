version https://git-lfs.github.com/spec/v1
oid sha256:66991e42613890ca16c01412caee3e026dce758fdfae6eb4c16e96c90ac25656
size 137439
