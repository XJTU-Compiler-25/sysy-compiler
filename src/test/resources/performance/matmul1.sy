int a[200][200];
int b[200][200];
int c[200][200];
const int MAX = 2147483647;

int main(){
    int n;
    int i;
    int j;
    int k;
    int sum = 0;

    i = 0;
    while(i<200)
    {
        n = getarray(a[i]);
        if(n!=200){
            return n;
        }
        i = i + 1;
    }

    starttime();

    i = 0;
    while(i<200){
        j = 0;
        while(j<200){
            b[i][j] = a[j][i];
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<200){
        j = 0;
        while(j<200){
            k = 0;
            int temp = 0;
            while(k<200){
                if(a[i][k]*b[k][j] % 2 == 0)
                    temp = temp + a[i][k]*b[k][j];
                k = k+1;
            }
            c[i][j] = temp;
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<200){
        j = 0;
        int temp = MAX;
        while(j<200){
            if(c[i][j]<temp)
            {
                temp = c[i][j];
            }
            j = j+1;
        }
        j = 0;
        while(j<200){
            c[i][j] = temp;
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<200){
        j = 0;
        int temp = MAX;
        while(j<200){
            c[i][j] = -c[j][i];
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<200){
        j = 0;
        int temp = MAX;
        while(j<200){
            sum = sum + c[i][j];
            j = j+1;
        }
        i = i+1;
    }

    stoptime();

    putint(sum);
    return 0;
}

