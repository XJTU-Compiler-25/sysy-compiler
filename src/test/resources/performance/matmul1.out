version https://git-lfs.github.com/spec/v1
oid sha256:f806b0c98bcf665df259ec7e13cd9a0ef5cd9e496d3ac0f599bfbe448610cfde
size 11
